service/history/transfer_queue_active_task_executor.go

```diff
@@ -928,6 +938,16 @@ func (t *transferQueueActiveTaskExecutor) processStartChildExecution(
 	t.logger.Debug("Child Execution started successfully",
 		tag.WorkflowID(attributes.WorkflowId), tag.WorkflowRunID(childRunID))
 
+	// if shouldTerminateAndStartChild is true, it means the child was started after the reset point and we attempted to terminate it before starting a new one.
+	// We should update the parent execution info to reflect that the child was potentially terminated and started.
+	if shouldTerminateAndStartChild {
+		childrenInitializedPostResetPoint := executionInfo.ChildrenInitializedPostResetPoint
+		childrenInitializedPostResetPoint[resetChildID] = &persistencespb.ResetChildInfo{
+			ShouldTerminateAndStart: false,
```

Comment:
do we ever clean up this field?

Reply:
I don't have any plans of removing this data yet. We need it to prevent forwarded results from old children. So, after the child completes we can delete this data.

Reply:
Please track the deletion work.

Reply:
PR coming up.

Reply:
https://github.com/temporalio/temporal/pull/7295